core/src/main/scala/kafka/server/AdminManager.scala

```diff
@@ -355,8 +355,8 @@ class AdminManager(val config: KafkaConfig,
       def createResponseConfig(configs: Map[String, Any],
                                createConfigEntry: (String, Any) => DescribeConfigsResponseData.DescribeConfigsResourceResult): DescribeConfigsResponseData.DescribeConfigsResult = {
         val filteredConfigPairs = configs.filter { case (configName, _) =>
-          /* Always returns true if configNames is None */
-          resource.configurationKeys.asScala.forall(_.contains(configName))
+          /* Always returns true if configurationKeys is null */
+          if (resource.configurationKeys != null) resource.configurationKeys.asScala.forall(_.contains(configName)) else true
```

Comment:
Since null means "all the keys" perhaps it would be better to move the `if` to the topic level to avoid an unneccessary traversal in the null case:

```scala
        val filteredConfigPairs = if (resource.configurationKeys == null)
          configs.toBuffer
        else 
          configs.filter { case (configName, _) =>
            /* Always returns true if configNames is None */
            resource.configurationKeys.asScala.forall(_.contains(configName))
          }.toBuffer
```

Reply:
good suggestion. Thanks.